students/kristianjf/lesson10/mailroom_6.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env python3
+import sys
+from functools import reduce
+'''
+Goal:
+You work in the mail room at a local charity. Part of your job is to write \
+incredibly boring, repetitive emails thanking your donors for their generous \
+gifts. You are tired of doing this over and over again, so you’ve decided to \
+let Python help you out of a jam and do your work for you.
+'''
+'''
+It should have a data structure that holds a list of your donors and a history of \
+the amounts they have donated. This structure should be populated at first with at \
+least five donors, with between 1 and 3 donations each.
+'''
+class Donor:
+    def __init__(self, name, donations):
+        self._name = name
+        self._normalized_name = ''
+        self._donations = donations if isinstance(donations, list) else [donations]
+    def __lt__(self, other):
+        return sum(self.donations) < sum(other.donations)
+    @property
+    def name(self):
+        return self._name
+    @property
+    def donations(self):
+        return self._donations
+    @property
+    def normalized_name(self):
+        'Validate and Return Name Input'
+        full_name_cap = ''
+        try:
+            for name in self.name.split():
+                assert name.isalpha()
```

Comment:
you don't want an assertion here -- asserts are for testing.

somethign like:

```
if not name.isapha():
    raise ValueError(names can only have alphabetical characters")
```

but this is not where you should test validity anyway :-)